src/collision.cpp

```diff
@@ -399,27 +408,24 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 		}
 	}
 
-	/*
-		Collect object boxes in movement range
-	*/
+	// Collect object boxes in movement range
 	if (collide_with_objects) {
-		add_object_boxes(env, box_0, dtime, *pos_f, *speed_f, self, cinfo);
+		add_object_boxes(env, box_0, dtime, *pos_f, aspeed_f, self, cinfo);
 	}
 
-	/*
-		Collision detection
-	*/
+	// Sort boxes by distance once, as we are more likely to collide with near boxes
+	v3f center_0 = box_0.getCenter();
+	for (auto box = cinfo.begin(); box != cinfo.end(); box++) {
+		box->distance = (box->box.getCenter() - center_0).getLengthSQ();
+	}
+	std::sort(cinfo.begin(), cinfo.end());
```

Comment:
It is not only about run time; but also to have the collision code prefer the closest boxes. Consider standing on the floor inbetween of multiple nodes, this will prefer reporting the one directly underneath, whereas without sorting it will likely report the one with the smallest coordinate.

I guess the sorting is somewhat expensive because the array stores structs, not pointers. Maybe we can insert them into the structure in a more ordered fashion in the first place, to reduce the sorting cost this way, what do you think?

Reply:
If memory moves are indeed the issue, then a `std::list` container should avoid that.

> but also to have the collision code prefer the closest boxes

They're already being sorted manually by `axisAlignedCollision` below.

Reply:
Collision time very frequently has ties.

Reply:
But the sorting is a separate commit, and it should work to omit it.